pkg/discovery/transport/client/client.go

```diff
@@ -99,33 +99,50 @@ func (c *Client) GetOut() chan *pb.Event {
 
 // Connect dials the server and returns a connection.
 func (c *Client) Connect() (*grpc.ClientConn, error) {
-	conn, err := grpc.Dial(c.conf.Host+":"+c.conf.Port, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithTimeout(c.conf.Timeout))
+	ctx, cancelConnect := context.WithTimeout(context.Background(), c.conf.ConnectTimeout)
+	defer cancelConnect()
+	conn, err := grpc.DialContext(ctx, c.conf.Host+":"+c.conf.Port, grpc.WithBlock(), grpc.WithInsecure())
 	if err != nil {
 		c.conf.Logger.Error("error establishing a gRPC connection")
```

Comment:
Shouldn't we log the error as well here?

Reply:
done